codegen/server/client_cli.go

```diff
@@ -0,0 +1,691 @@
+		Example string
+		// MultipartFuncName is the function name used to render a multipart request encoder.
+		MultipartFuncName string
+		// MultipartFuncName is the variable name used to render a multipart request encoder.
+		MultipartVarName string
+	}
+
+	// FlagData contains the data needed to render a flag
+	FlagData struct {
+		// Name is the name of the flag, e.g. "list-vintage"
+		Name string
+		// VarName is the name of the flag variable, e.g. "listVintage"
+		VarName string
+		// Type is the type of the flag, e.g. INT
+		Type string
+		// FullName is the flag full name e.g. "storageAddVintage"
+		FullName string
+		// Description is the flag help text.
+		Description string
+		// Required is true if the flag is required.
+		Required bool
+		// Example returns a JSON serialized example value.
+		Example string
+	}
+
+	// BuildFunctionData contains the data needed to render a build function
+	BuildFunctionData struct {
+		// Name is the build payload function name.
+		Name string
+		// Description describes the payload function.
+		Description string
+		// ActualParams is the list of passed build function parameters.
+		ActualParams []string
+		// FormalParams is the list of build function formal parameter
+		// names.
+		FormalParams []string
+		// ServiceName is the name of the service.
+		ServiceName string
+		// MethodName is the name of the method.
+		MethodName string
+		// ResultType is the fully qualified payload type name.
+		ResultType string
+		// ReturnTypeName is the same as ResultType
+		ReturnTypeName string
+		// Fields describes the payload fields.
+		Fields []*FieldData
+		// PayloadInit contains the data needed to render the function
+		// body.
+		PayloadInit *PayloadInitData
+		// CheckErr is true if the payload initialization code requires
+		// an "err error" variable that must be checked.
+		CheckErr bool
+	}
+
+	// FieldData contains the data needed to render a field
+	FieldData struct {
+		// Name is the field name, e.g. "Vintage"
+		Name string
+		// VarName is the name of the local variable holding the field
+		// value, e.g. "vintage"
+		VarName string
+		// TypeName is the name of the type.
+		TypeName string
+		// Init is the code initializing the variable.
+		Init string
+		// Pointer is true if the variable needs to be declared as a
+		// pointer.
+		Pointer bool
+	}
+
+	// PayloadInitData contains the data needed to render payload initializers
+	PayloadInitData struct {
+		Code                string
+		ReturnTypeAttribute string
+		ReturnIsStruct      bool
+		ReturnTypeName      string
+		Args                []*PayloadInitArgData
+	}
+
+	// PayloadInitArgData contains the data needed to render payload initlization args
+	PayloadInitArgData struct {
+		Name      string
+		FieldName string
+	}
+)
+
+// BuildCommandData builds the data needed by the templates to render the CLI
+// parsing of the service command.
+func BuildCommandData(name, description, pkgName string, needsStream bool) *CommandData {
```

Comment:
I think `BuildCommandData` can have the following signature
```
func BuildCommandData(data *service.Data) *CommandData
```
Both HTTP and gRPC CLI code is using `service.Data` to figure out the name, description, pkgName, etc anyways. We can figure out the `NeedStream` field in the common code (although it doesn't have any effect for gRPC).